contracts/base/Dispatcher.sol

```diff
@@ -33,171 +37,217 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks,
+                        bytes calldata data = inputs.toBytes(6); // PermitSingle takes first 6 slots (0..5)
+                        PERMIT2.permit(lockedBy, permitSingle, data);
+                    } else if (command == Commands.WRAP_ETH) {
+                        // equivalent: abi.decode(inputs, (address, uint256))
+                        address recipient;
+                        uint256 amountMin;
+                        assembly {
+                            recipient := calldataload(inputs.offset)
+                            amountMin := calldataload(add(inputs.offset, 0x20))
+                        }
+                        Payments.wrapETH(map(recipient), amountMin);
+                    } else if (command == Commands.UNWRAP_WETH) {
+                        // equivalent: abi.decode(inputs, (address, uint256))
+                        address recipient;
+                        uint256 amountMin;
+                        assembly {
+                            recipient := calldataload(inputs.offset)
+                            amountMin := calldataload(add(inputs.offset, 0x20))
+                        }
+                        Payments.unwrapWETH9(map(recipient), amountMin);
+                    } else if (command == Commands.PERMIT2_TRANSFER_FROM_BATCH) {
+                        (IAllowanceTransfer.AllowanceTransferDetails[] memory batchDetails) =
+                            abi.decode(inputs, (IAllowanceTransfer.AllowanceTransferDetails[]));
+                        permit2TransferFrom(batchDetails, lockedBy);
+                    } else if (command == Commands.BALANCE_CHECK_ERC20) {
+                        // equivalent: abi.decode(inputs, (address, address, uint256))
+                        address owner;
+                        address token;
+                        uint256 minBalance;
+                        assembly {
+                            owner := calldataload(inputs.offset)
+                            token := calldataload(add(inputs.offset, 0x20))
+                            minBalance := calldataload(add(inputs.offset, 0x40))
+                        }
+                        success = (ERC20(token).balanceOf(owner) >= minBalance);
+                        if (!success) output = abi.encodePacked(BalanceTooLow.selector);
+                    } else {
+                        // placeholder area for command 0x0f
+                        revert InvalidCommandType(command);
+                    }
                 }
-                // 0x08 <= command < 0x10
+                // 0x10 <= command < 0x18
             } else {
-                if (command == Commands.V2_SWAP_EXACT_IN) {
-                    // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
-                    address recipient;
-                    uint256 amountIn;
-                    uint256 amountOutMin;
-                    bool payerIsUser;
+                if (command == Commands.ERC721_PERMIT) {
```

Comment:
why cant this command just be joined with the `V3_POSM_CALL` command? it just decodes the parameters, and then passes them straight through to the position manager right?

Reply:
hm but v3 position manager call always takes in tokenId as the first parameter, permit needs to take spender as the first parameter

Reply:
yeah good point 👍

Reply:
i do think theres no need to decode all the parameters though. It could just have data, check the selector is for permit, and then call without decoding everything?

Reply:
oh I see because we wouldn't have to check isAuthorizedForToken if the action is erc721permit